.github/workflows/codeql.yml

```diff
@@ -1,17 +1,98 @@
 # Workflow for advanced CodeQL setup used for scanning Java/JavaScript/TypeScript/Vue/Python based source files
 name: "CodeQL Advanced"
+env:
+  # Whether to analyze Java code or not (only set to true if repo has Java source code)
+  analyze-java: true
+  # Build mode to use for analysis of Java code (e.g. none, autobuild, manual)
+  java-buildmode: "autobuild"
+  # Temurin JDK version to use for autobuild (only when java-buildmode is set to autobuild)
+  java-version: "21"
+  # Whether to analyze JavaScript/TypeScript/Vue code or not (only set to true if repo has Javascript/Typescript/Vue source code)
+  analyze-javascript-typescript-vue: true
+  # Whether to analyze Python code or not (only set to true if repo has Python source code)
+  analyze-python: false
+  # Query set to use when analyzing the source code (e.g. default, security-extended, security-and-quality)
+  analysis-query: security-and-quality
 
 on:
   # Runs on pull requests and on pushes to main (in order to keep the regular scanning by GitHub working)
   pull_request:
   push:
     branches: [main]
 
+permissions:
+  pull-requests: read
+  security-events: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  codeql:
-    uses: it-at-m/.github/.github/workflows/codeql.yml@main
-    with:
-      analyze-java: true
-      analyze-javascript-typescript-vue: true
-      analyze-python: false
-      java-build-paths: "['./refarch-backend','./refarch-eai']"
+  check-changes:
+    name: Check changed files and directories
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      java: ${{ steps.filter.outputs.java }}
+      javascript-typescript-vue: ${{ steps.filter.outputs.javascript-typescript-vue }}
+      python: ${{ steps.filter.outputs.python }}
+    steps:
+      - name: Checkout repository
+        uses: it-at-m/lhm_actions/action-templates/actions/action-checkout@codeql
+      - name: Path Filter
+        id: filter
+        uses: it-at-m/lhm_actions/action-templates/actions/action-filter@codeql
+        with:
+          filters: |
+            java:
+              - '**/*.java'
+            javascript-typescript-vue:
+              - '**/*.js'
+              - '**/*.cjs'
+              - '**/*.mjs'
+              - '**/*.ts'
+              - '**/*.cts'
+              - '**/*.mts'
+              - '**/*.vue'
+            python:
+              - '**/*.py'
+  codeql-java:
+    name: Analyze Java source files
+    runs-on: ubuntu-latest
+    needs: check-changes
+
+    strategy:
+      fail-fast: false
+      matrix:
+        build-path: ["./refarch-backend", "./refarch-eai"] # JSON array formatted as string, contains the paths to the java projects to build
+    steps:
+      - uses: it-at-m/lhm_actions/action-templates/actions/action-maven-build@v1.0.3
+        if: env.analyze-java && (github.ref_name == 'main' || needs.check-changes.outputs.java == 'true')
```

Comment:
In comment https://github.com/it-at-m/refarch-templates/pull/860#discussion_r1983332377 actionlint marks env as not allowed. The comment was closed because it's outdated but it seems that the code itself didn't change.

Reply:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability dann die Variable komplett entfernen. Da man nicht auf sie zugreifen kann

Reply:
```suggestion
        if: (github.ref_name == 'main' || needs.check-changes.outputs.java == 'true')
```

Reply:
https://github.com/it-at-m/refarch-templates/pull/860#discussion_r2013524707

Reply:
ok, danke für die Info.